ee/src/main/java/org/jboss/as/ee/concurrent/ConcurrencyImplementationLoader.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.jboss.as.ee.concurrent;
+
+import org.jboss.as.ee.logging.EeLogger;
+import org.jboss.modules.Module;
+import org.jboss.modules.ModuleLoader;
+
+import java.util.ServiceLoader;
+
+/**
+ * ConcurrencyImplementation service loader.
+ * @author emartins
+ */
+class ConcurrencyImplementationLoader {
+
+    private ConcurrencyImplementationLoader() {
+    }
+
+    /**
+     *
+     * @return the implementation loaded using JBoss Module classloader, if found, otherwise using current thread's class loader
```

Comment:
Is the TCCL scenario for subsystem testing? If so it would be good to say that in the javadoc.

Reply:
fixed